### PR TITLE
addpkg(x11/xwallpaper): 0.7.5

### LINKED ISSUE #17729: [Package]: xwallpaper: wallpaper setting utility for X
### Why is it worth to add this package?

Hello,

This is a very minimal and easy to use standalone wallpaper setting utility.

no gui, no imageviewer, no live wallpapers. Just a cli utility to set your wallpaper.

present in the [rocks](https://suckless.org/rocks/#) category of suckless.org .

### Home page URL

https://github.com/stoeckmann/xwallpaper

### Source code URL

https://github.com/stoeckmann/xwallpaper

### Packaging policy acknowledgement

- [X] I certify that I have read [Termux Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy) and understand that my request will be denied in case of violation.

### Additional information

build steps:
1. `git clone https://github.com/stoeckmann/xwallpaper`
2. `autoreconfigure -fi`
3. patch the file (configure) with the following:
```
***************
*** 598,604 ****
  #
  # Initializations.
  #
- ac_default_prefix=/usr/local
  ac_clean_files=
  ac_config_libobj_dir=.
  LIBOBJS=
--- 598,604 ----
  #
  # Initializations.
  #
+ ac_default_prefix=/data/data/com.termux/files/usr
  ac_clean_files=
  ac_config_libobj_dir=.
  LIBOBJS=

```
4. `~ $ ./configure`
5. `~ $ make` 
6. `~ $ make clean install`

### PR DESCRIPTION
**Pull request by twaik**
Fixes #17729

### CODE REVIEW

**Comment by truboxl** [comment]
Seems pretty useless without any image support

**Comment by twaik** [comment]
Later I want to do some on-device checks with xfce, mate and openbox.

**Comment by twaik** [comment]
Works fine on my device when there is no DE running.

<details>

image

</details>

**Comment by twaik** [comment]
Added png, jpg and xpm support.

**Comment by twaik** [comment]
I'll merge it in 72 hours if nobody minds.
Or anyone else can merge it earlier.

File: x11-packages/xwallpaper/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/stoeckmann/xwallpaper
+TERMUX_PKG_DESCRIPTION="wallpaper setting utility for X"
+TERMUX_PKG_LICENSE="ISC"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=0.7.5
+TERMUX_PKG_SRCURL=https://github.com/stoeckmann/xwallpaper/archive/refs/tags/v0.7.5.tar.gz
```
Comment (truboxl) [inline x11-packages/xwallpaper/build.sh:6]: Do not hard code version

File: x11-packages/xwallpaper/build.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+TERMUX_PKG_SHA256=d64b8bae1700835d1c0996b28ff0e9d4a93ead4f8698bbdb6acc19150537aa23
+TERMUX_PKG_DEPENDS="libpixman, xcb-util-image, libjpeg-turbo, libpng, libxpm"
+
+termux_step_pre_configure() {
+	export PKG_CONFIG_PATH=/data/data/com.termux/files/usr/lib/pkgconfig
```
Comment (truboxl) [inline x11-packages/xwallpaper/build.sh:11]: I do not believe this is needed. Also why the hardcode prefix?

File: x11-packages/xwallpaper/build.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/stoeckmann/xwallpaper
+TERMUX_PKG_DESCRIPTION="wallpaper setting utility for X"
+TERMUX_PKG_LICENSE="ISC"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=0.7.5
+TERMUX_PKG_SRCURL=https://github.com/stoeckmann/xwallpaper/archive/refs/tags/v0.7.5.tar.gz
+TERMUX_PKG_SHA256=d64b8bae1700835d1c0996b28ff0e9d4a93ead4f8698bbdb6acc19150537aa23
+TERMUX_PKG_DEPENDS="libpixman, xcb-util-image, libjpeg-turbo, libpng, libxpm"
```
Comment (truboxl) [inline x11-packages/xwallpaper/build.sh:8]:
```suggestion
TERMUX_PKG_DEPENDS="libjpeg-turbo, libpixman, libpng, libxcb, libxpm, xcb-util, xcb-util-image"
```

**Review by truboxl — CHANGES_REQUESTED** [review]

File: x11-packages/xwallpaper/build.sh (hunk 2 above)
Comment (twaik) [reply x11-packages/xwallpaper/build.sh:11]: I was invoking `./build-package` with `-s` on clean environment and wondered why it does not work. Imma stupid, sorry.

**Comment by twaik** [comment]
Sorry, was a bit tired. Fixed this.

**Review by truboxl — APPROVED** [review]

**Pull request merged**